### DISCUSSION TITLE
Why write buffer before jump?

### DISCUSSION THREAD

**Discussion by dane-harnett** [nvimdev/lspsaga.nvim#857] [Q&A] [answered]
Hi, I noticed that my setup was formatting my buffers when using goto definition.
I have an autocmd on BufWritePre that runs `vim.lsp.buf.format()`

I looked at the code and here https://github.com/glepnir/lspsaga.nvim/blob/main/lua/lspsaga/definition.lua#L290
it is writing the buffer before jumping to the definition.

What is the reason for this?

Can this be optional behaviour?

Also I see here https://github.com/glepnir/lspsaga.nvim/blob/cc9c23488d01b2aa72961ed71b955073acbfb5e2/lua/lspsaga/finder.lua#L788-L790
that it checks if the buffer is modified before writing... could this also apply to goto definition?

I'm happy to raise PRs when I understand the reasons :)

Thank you for your hard work

**Comment by glepnir** [comment, marked as answer]
it work for the definition in same file.  there can be `vim.cmd('noautocmd write')`

**Reply by dane-harnett** [reply]
Thank you that makes sense.
So as I understand it the current buffer must be written so that if the definition exists in the current buffer it will jump to the right spot.

Is it acceptable to check first if the current buffer has been edited before writing?

also, is it acceptable to also check first if the current buffer is the jump destination?

I don't think I would want to disable the autocmd because if the current buffer needs to be saved I would like for it to be formatted too.

What do you think? I can try to create a PR this weekend if you agree.

**Reply by glepnir** [reply]
PR welcome
